### DISCUSSION TITLE
Sizing and Quantization

### DISCUSSION THREAD

**Discussion by wilsonlimaneto** [milvus-io/milvus#24662] [Q&A and General discussion]
Hello, well since I am getting OOM errors with 7 million vectors, I started to research more about scaling Milvus or some other options. Right now I have a transformer which yields 1024 dim vectors dtype float. I then, turn those into ndarray to save inside milvus thru haystack and save them. But, once I need to save 500 million vectors, and I need RAM memory for all that, does milvus already do any quantization in any way? It is impossible to use 2000Gb RAM so... Any advice on how should I proceed on sizing or any idea?1 In the meanwhile I will keep on researching thaks

**Comment by xiaofan-luan** [comment]
> million vectors, and I need RAM memory for all that, does milvus already do any quantization in any way? It is impossible to use 2000Gb RAM so... Any advice on how should I proceed on sizing or any idea?1 In the meanwhile I will keep on researching thaks

1 million takes around 5-6GB memories. for 500 million, I would expect something more than 2TB memories.

When do OOM begin, and what is the Milvus version?

Another solution is to use IVF_SQ8, where you only need 1/4 of your memory but the recall will be lower than than IVF.
